metricchecks/check_impl.go

```diff
@@ -179,21 +185,21 @@ func (c *checker) InsertMetricValuesFromContext(m *metrics.MetricContext) error
 	for metricName, metric := range m.Gauges {
 		name := strings.Replace(metricName, ".", "_", -1) + "_value"
 		c.sc.Insert(types.NewConst(0, c.pkg, name,
-			types.New("float64"), exact.MakeFloat64(metric.Get())))
+			types.Typ[types.Float64], exact.MakeFloat64(metric.Get())))
 		sname := name + "_string"
 		c.sc.Insert(types.NewConst(0, c.pkg, sname,
-			types.New("string"), exact.MakeString(fmt.Sprintf("%0.2f", metric.Get()))))
+			types.Typ[types.String], exact.MakeString(fmt.Sprintf("%0.2f", metric.Get()))))
 	}
 	for metricName, metric := range m.Counters {
 		name := strings.Replace(metricName, ".", "_", -1) + "_current"
 		c.sc.Insert(types.NewConst(0, c.pkg, name,
-			types.New("float64"), exact.MakeUint64(metric.Get())))
+			types.Typ[types.Float64], exact.MakeUint64(metric.Get())))
```

Comment:
@syamp 
I saw that the old code created a float64 type but appears to compare it to a uint64, so I just mirrored this difference. Should the type be float64 or uint64?